### ISSUE TITLE
Add separate commands for changing the height and width of a view (not both)

### ISSUE THREAD

**Issue by jqmp** [microsoft/vscode#102954]
I would like to have a command that increases the height of the current view, but not its width. The closest command I can find is `workbench.action.increaseViewSize`, which increases both the height and the width at once.

My use case: I like to have a large number of splits, both horizontal and vertical. This allows me to quickly switch among a group of files without using the mouse or searching by filename. However, I usually want my current view to be maximized vertically (so I can see more code) but not horizontally (because my files have fixed line lengths, so there's no point in making the view any wider).

Essentially I want to reproduce the Vim command `Ctrl+W` `_` ("maximize the current view's height"). Having a command to do this directly would be nice, but a vertical-only equivalent of `workbench.action.increaseViewSize` would also work and might be more general. The current behavior of `workbench.action.increaseViewSize` seems like it really only makes sense if you have horizontal *or* vertical splits, but not both.

After skimming the code, this seems like it might not be hard to implement, so if the maintainers support the idea and time permits I might be able to submit a PR.

Thanks!

**Comment by isidorn** [comment]
I tried to verify this, but reopening due to the following:
* I could not make this commands to work in general. They only work if I have multiple editors in a split view. If I have an editor and at the bottom the panel, the editor will not be resized
* These commands are only active when an editor is focused. Should the commands have a different name then?
* I do not really inderstand why do we need new commands compared to the already present

**Comment by sbatten** [comment]
@isidorn sorry about the confusion here. when I tested, I thought the context key was working properly because the commands were still accessible via recent commands in the command palette, but we cannot use the focus context key for the cmd palette visibility.

I've updated the commands to be titled referencing the editor instead as you mention, and they should now work even if the editor is not focused.

the new commands especially useful any time the editor is split in both directions. currently, in a two by two editor situation, the increase view size command makes an individual editor group bigger in both directions when it is most likely true that you'd rather increase the height only.

**Comment by jqmp** [comment]
Thank you for adding this, @sbatten!

**Comment by davidysoards** [comment]
I have these commands mapped as follows:
```
    {
      "key": "ctrl+=",
      "command": "workbench.action.increaseViewSize",
      "when": "editorFocus"
    },
    {
      "key": "ctrl+-",
      "command": "workbench.action.decreaseViewSize",
      "when": "editorFocus"
    },
```
With 2 or more tabs open horizontally and the terminal across the bottom, prior to v1.51.1 when I increased or decreased it would affect only the width of my active editor tab. Which makes perfect sense.

But now it also affects the height. Or more accurately is affects the height of the terminal. This is super annoying.  Was this an intentional change?

The terminal should be treated as a fixed overlay.

I can open a new issue for this if necessary.
